client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/transaction/XaModeTransactionTable.java

```diff
@@ -133,8 +139,13 @@ public void commit(Xid xid, boolean onePhaseCommit) throws XAException {
          if (trace) {
             log.tracef("XaResource.commit(%s, %s)", xid, onePhaseCommit);
          }
-         assertStartInvoked();
-         assertSameXid(xid, XAException.XAER_INVAL);
+         if (currentXid == null) {
+            //no transaction running. we are doing some recovery work
```

Comment:
"doing some recovery work" sounds too vague to me, like we could be doing more than just committing the tx

Reply:
recovery only commits or rollbacks in-doubt and prepared transactions. there is nothing else to do.